Fable.Remoting.MsgPack/Read.fs

```diff
@@ -23,36 +27,43 @@ let interpretStringAs (typ: Type) (str: string) =
         FSharpValue.MakeUnion (case, [||], true)
 #endif
 
-let inline interpretIntegerAs typ n =
+let inline interpretIntegerAs (typ: Type) n =
+#if !FABLE_COMPILER
     if typ = typeof<Int32> then int32 n |> box
     elif typ = typeof<Int64> then int64 n |> box
     elif typ = typeof<Int16> then int16 n |> box
     elif typ = typeof<UInt32> then uint32 n |> box
     elif typ = typeof<UInt64> then uint64 n |> box
     elif typ = typeof<UInt16> then uint16 n |> box
     elif typ = typeof<TimeSpan> then TimeSpan (int64 n) |> box
-#if FABLE_COMPILER
-    elif typ.FullName = "Microsoft.FSharp.Core.int16`1" then int16 n |> box
-    elif typ.FullName = "Microsoft.FSharp.Core.int32`1" then int32 n |> box
-    elif typ.FullName = "Microsoft.FSharp.Core.int64`1" then int64 n |> box
-#endif
     elif typ = typeof<byte> then byte n |> box
     elif typ = typeof<sbyte> then sbyte n |> box
-#if !FABLE_COMPILER
     elif typ.IsEnum then Enum.ToObject (typ, int64 n)
 #else
+    if Object.ReferenceEquals (typ, typeof<Int32>) then int32 n |> box
+    elif typ.FullName = "System.Int64" then int64 n |> box
+    elif Object.ReferenceEquals (typ, typeof<Int16>) then int16 n |> box
+    elif Object.ReferenceEquals (typ, typeof<UInt32>) then uint32 n |> box
+    elif typ.FullName = "System.UInt64" then uint64 n |> box
+    elif Object.ReferenceEquals (typ, typeof<UInt16>) then uint16 n |> box
```

Comment:
![msedge_0qP8jy3Tlt](https://user-images.githubusercontent.com/5063478/145473986-da7a2ca8-8265-407a-b744-a1c8db85e738.png)

This would suggest I can take a shortcut with reference equality, but as I said, not sure if I can always rely on it.

Reply:
Oh wow! This is super! Go Rangers, go! :) I'll take the patch it for a spin as soon as I can!

Reply:
@kerams Sorry for replying right. Your assumption is totally right. The fact that `typeof<int>` compiles to a static `Reflection.ts/Type` instance can be considered an implementation detail but the following code also evals to true in dotnet so if you want you can PR a test like this to Fable to make sure we don't have regressions in the future:

```fsharp
let test(t: Type) =
  obj.ReferenceEquals(t, typeof<int>) |> printfn "%b"

let inline test2(x: 'T) =
  x.GetType() |> test

test2 4
```

Reply:
Thank you, Alfonso, I think I'll do that.